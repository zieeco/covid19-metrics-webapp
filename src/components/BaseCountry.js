import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getNigeriaDetails } from '../redux/countries/baseCountry';

const BaseCountry = () => {
  const myUrl = 'https://raw.githubusercontent.com/rachidelaid/worldMaps/main/maps/ng/vector.svg';
  const myCountry = useSelector((state) => state.baseCountryReducer);
  const dispatch = useDispatch();

  useEffect(() => {
    if (myCountry.length === 0) {
      dispatch(getNigeriaDetails());
    }
  }, []);

  return (
    <div className="base-container flex">
      {myCountry.map((country) => {
        const floatNumber = parseFloat(country.today_confirmed);
        const floatNumberFormat = floatNumber.toLocaleString('en');
        return (
          <div key={country.id} className="base-ng flex">
            <div className="ng-box flex">
              <img src={myUrl} alt={`${country.name}'s map`} className="ng-img map-images" />
            </div>
            <div className="ng-body flex">
              <p className="fw fs-1">{country.name.toUpperCase()}</p>
              <p className="fw fs-1">{floatNumberFormat}</p>
            </div>
          </div>
        );
      })}
    </div>
  );
};

export default BaseCountry;
