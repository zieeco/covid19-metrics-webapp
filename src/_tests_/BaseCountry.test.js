import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import BaseCountry from '../components/BaseCountry';
import store from '../redux/configureStore';

const BaseCountryProvider = () => (
  <Provider store={store}>
    <BaseCountry />
  </Provider>
);

describe('test for BaseCountry component', () => {
  test('should render correctly', () => {
    const RenderedComponent = renderer.create(<BaseCountryProvider />).toJSON();
    expect(RenderedComponent).toMatchSnapshot();
  });

  test('reders from the api', async () => {
    render(<BaseCountryProvider />);
    const imageText = await screen.findByAltText(/nigeria map/i);
    console.log('the image text', typeof imageText);
    expect(imageText).toBeInTheDocument();
  });
});
