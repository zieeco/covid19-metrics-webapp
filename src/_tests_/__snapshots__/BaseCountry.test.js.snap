// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for BaseCountry component should render correctly 1`] = `
<div
  className="base-container flex"
/>
`;
